google-cloud-spanner/lib/google/cloud/spanner/client.rb

```diff
@@ -471,12 +471,11 @@ def transaction &block
             tx = Transaction.from_grpc(tx_grpc, session)
             begin
               block.call tx
-            rescue Google::Cloud::AbortedError
-              # TODO: retrieve delay from ABORTED error
-              # Retry the entire transaction
-              tx2_grpc = @project.service.begin_transaction session.path
-              tx2 = Transaction.from_grpc(tx2_grpc, session)
-              block.call tx2
+            rescue Google::Cloud::AbortedError => err
+              sleep delay_from_aborted(err)
+              tx_grpc = @project.service.begin_transaction session.path
+              tx = Transaction.from_grpc(tx_grpc, session)
+              retry
```

Comment:
It should be user configurable with some default value. Foe eg if user sets it to 2 minutes, you keep on trying till then. Also for the outgoing commit rpc, you should set the timeout to be whatever is remaining. So lets say 1 minute out of those 2 mintues have elapsed, then you set the timeout for the outgoing commit rpc to 1 minute.